src/runtime-rs/crates/resource/src/network/network_model/route_model.rs

```diff
@@ -0,0 +1,105 @@
+// Copyright (c) 2019-2022 Alibaba Cloud
+// Copyright (c) 2019-2022 Ant Group
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+use std::process::Command;
+
+use anyhow::{anyhow, Context, Result};
+use async_trait::async_trait;
+
+use super::{NetworkModel, NetworkModelType};
+use crate::network::NetworkPair;
+
+#[derive(Debug)]
+pub(crate) struct RouteModel {}
+
+impl RouteModel {
+    pub fn new() -> Result<Self> {
+        Ok(Self {})
+    }
+}
+
+#[async_trait]
+impl NetworkModel for RouteModel {
+    fn model_type(&self) -> NetworkModelType {
+        NetworkModelType::Route
+    }
+
+    async fn add(&self, pair: &NetworkPair) -> Result<()> {
+        let tap_name = &pair.tap.tap_iface.name;
+        let virt_name = &pair.virt_iface.name;
+
+        //setup rule
+        let output = Command::new("ip")
+            .args(&[
+                "rule", "add", "pref", "10", "from", "all", "lookup", "local",
+            ])
+            .output()
+            .context("ip rule add pref 10 from all lookup local")?;
+        if !output.status.success() {
+            return Err(anyhow!("{}", String::from_utf8(output.stderr)?));
+        }
+
+        let output = Command::new("ip")
+            .args(&["rule", "del", "pref", "0", "from", "all"])
+            .output()
+            .context("ip rule del pref 0 from all")?;
+        if !output.status.success() {
+            return Err(anyhow!("{}", String::from_utf8(output.stderr)?));
+        }
+
+        let output = Command::new("ip")
+            .args(&["rule", "add", "pref", "5", "iif"])
+            .arg(&virt_name)
+            .args(&["table", "10"])
+            .output()
+            .context(format!("ip rule add pref 5 iif {} table 10", &virt_name))?;
+        if !output.status.success() {
+            return Err(anyhow!("{}", String::from_utf8(output.stderr)?));
+        }
+
+        // set route
+        let output = Command::new("ip")
+            .args(&["route", "replace", "default", "dev"])
+            .arg(&tap_name)
+            .args(&["table", "10"])
+            .output()
+            .context("ip route replace default dev $TAP table 10")?;
+        if !output.status.success() {
+            return Err(anyhow!("{}", String::from_utf8(output.stderr)?));
+        }
+
+        let output = Command::new("ip")
+            .args(&["neigh", "replace"])
+            .arg(pair.virt_iface.addrs[0].addr.to_string())
+            .arg("lladdr")
+            .arg(&pair.virt_iface.hard_addr)
+            .arg("dev")
+            .arg(&tap_name)
+            .output()
+            .context("ip neigh replace $IP lladdr $MAC dev $TAP")?;
+        if !output.status.success() {
+            return Err(anyhow!("{}", String::from_utf8(output.stderr)?));
+        }
+
+        //change sysctl for tap0_kata
```

Comment:
space is missing after //

Reply:
Thanks for your review, already add space.